x-pack/plugin/identity-provider/src/main/java/org/elasticsearch/xpack/idp/saml/authn/SuccessfulAuthenticationResponseMessageBuilder.java

```diff
@@ -215,4 +212,16 @@ private Status buildStatus() {
 
         return status;
     }
+
+    private NameID buildNameId(SamlServiceProvider serviceProvider, @Nullable SamlAuthenticationState authnState) {
+        final NameID nameID = samlFactory.object(NameID.class, NameID.DEFAULT_ELEMENT_NAME);
+        if (authnState != null && authnState.getRequestedNameidFormat() != null) {
+            nameID.setFormat(authnState.getRequestedNameidFormat());
+        } else {
+            nameID.setFormat(serviceProvider.getAllowedNameIdFormat() != null ? serviceProvider.getAllowedNameIdFormat() :
+                idp.getServiceProviderDefaults().nameIdFormat);
+
+        }
+        return nameID;
```

Comment:
No `Value` for the NameID? If that's intentional, can we add a `TODO` here to set it?